src/sql/workbench/contrib/notebook/browser/outputs/gridOutput.component.ts

```diff
@@ -144,8 +146,43 @@ export class GridOutputComponent extends AngularDisposable implements IMimeCompo
 			this._layoutCalledOnce = true;
 		}
 	}
+	reorderGridData(source: IDataResource) {
+		let rowIndex = 0;
+		// Get schema list of grid
+		const columnNames: string[] = source.schema.fields.map(field => field.name);
+		if (source.data.length > 0) {
+			// Check to see if data source is ordered properly based on schema
+			let rowKeys = Object.keys(source.data[0]);
+			if (!equals(columnNames, rowKeys)) {
+				// SQL notebooks does not use columnName as key (instead uses indices)
+				// Indicies indicate the row is ordered properly
+				// We check the data to know if it is in index form or string form
+				for (let index of rowKeys) {
```

Comment:
index doesn't really make sense here since it's not always an index - rowKey seems fine